### PR TITLE
Now uses SQLAlchemy event system

### PR DESCRIPTION
**Pull request by jtl999**

### CODE REVIEW

**Comment by bkabrda** [comment]
Thanks a bunch for the PR. On the first look, it seems to be ok. I'll try to do some testing this afternoon and merge if I don't find any issues. One minor thing that I'd like to ask you to fix is putting 1 empty line between the defined methods `after_*`.

File: flask_whooshee.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 import whoosh.index
 import whoosh.qparser
 
-from flask_sqlalchemy import models_committed, BaseQuery
-from sqlalchemy import text
+from flask import current_app
```
Comment (bkabrda) [inline flask_whooshee.py]: Is this import really necessary? I don't see it used anywhere else in the patch.

File: flask_whooshee.py [hunk 2]
```diff
@@ -265,9 +267,15 @@ def create_index(self, wh):
                 os.makedirs(index_path)
             index = whoosh.index.create_in(index_path, wh.schema)
         wh.index = index
-
-    def on_commit(self, app, changes):
-        """Method that gets connected to flask_sqlalchemy.models_committed, where it serves
+    def after_insert(self, mapper, connection, target):
```
Comment (bkabrda) [inline flask_whooshee.py]: Can you please put newlines between the newly defined methods? Just so the code reads better.

**Comment by jtl999** [comment]
Completed the changes

**Comment by jtl999** [comment]
On a sidenote: in test.py shouldn't `SQLALCHEMY_TRACK_MODIFICATIONS` be set to `False` as we are no longer using the old models_committed system? 

Although it's just a warning.

**Comment by bkabrda** [comment]
I'll set `SQLALCHEMY_TRACK_MODIFICATIONS` to `False` in a separate commit. Again, thanks a lot for sending this PR!

**Comment by bkabrda** [comment]
Hmm, I just noticed that there's `event.listen(model, 'after_delete', self.after_insert)` - shouldn't the last argument be `self.after_delete`, @jtl999?

**Comment by bkabrda** [comment]
Hehe, so this is interesting. There actually should be after_delete, but due to the way the `whooshee_search` method works, this doesn't matter - whoosh index finds the match since the object is not actually deleted, but there is no DB object with that ID, so it still returns no results. I'm going to fix this and extend `test_delete` to make sure that the search term has actually been deleted from the result :)

**Comment by bkabrda** [comment]
Fixed in https://github.com/bkabrda/flask-whooshee/commit/620a6f1588691f50a463b4ffb1d5a07d1d2b4194

**Comment by jtl999** [comment]
Ah, I noticed the discrepancy but noticed the same behavior as you.

**Pull request merged**